CARDS-1700: Add the ability for developers and testers to view the emails that are sent by CARDS

IntegralProgrammer opened the pull request:
This PR implements CARDS-1700 by providing the `cards/postfix-docker` Docker Image that listens for SMTP and SMTPS connections and writes the received email messages to a UNIX Mbox file where these messages can be viewed with a desktop mail client such as _GNOME Evolution_. This Docker container can run either independently and listen for messages from a CARDS instance started with `start_cards.sh` or it can be run in a Docker Compose service cluster and listen for messages from a CARDS instance within that cluster.

To test, follow the instructions in `Utilities/Development/EmailServer/README.md`.

IntegralProgrammer committed:
CARDS-1700: Add the ability for developers and testers to view the emails that are sent by CARDS

Add the cards/postfix-docker container and instructions for using it with a standalone CARDS instance started with the start_cards.sh command

IntegralProgrammer committed:
CARDS-1700: Add the ability for developers and testers to view the emails that are sent by CARDS

Fixes to README.md

IntegralProgrammer committed:
CARDS-1700: Add the ability for developers and testers to view the emails that are sent by CARDS

Fixes to README.md

IntegralProgrammer committed:
CARDS-1700: Add the ability for developers and testers to view the emails that are sent by CARDS

Support for using with generate_compose_yaml.py by using the --smtps_test_container flag

IntegralProgrammer committed:
CARDS-1700: Add the ability for developers and testers to view the emails that are sent by CARDS

Accept emails from all domains, not just fakemailhost

sdumitriu commented on Utilities/Development/EmailServer/Dockerfile:
```diff
@@ -0,0 +1,28 @@
+# specific language governing permissions and limitations
+# under the License.
+
+FROM debian:11
```
Can `11-slim` be used instead?

IntegralProgrammer replied:
Implemented by commit `1d1988c`. Reduces the built Docker image size from 215MB to 168MB.

sdumitriu commented on distribution/docker_entry.sh:
```diff
@@ -177,6 +177,14 @@ then
   SMTPS_VARIABLES="$SMTPS_VARIABLES -V emailnotifications.smtps.checkserveridentity=false"
 fi
 
+if [[ "$SMTPS_LOCAL_TEST_CONTAINER" == "true" ]]
```
I would rather have a standard directory for various certs and load all of them instead of one by one, conditioned by different feature flags.

IntegralProgrammer replied:
Implemented this change in commit `c8a7a3c` :heavy_check_mark:

IntegralProgrammer committed:
CARDS-1700: Add the ability for developers and testers to view the emails that are sent by CARDS

Change the base Docker image for cards/postfix-docker from debian:11 to the smaller debian:11-slim

IntegralProgrammer committed:
CARDS-1700: Add the ability for developers and testers to view the emails that are sent by CARDS

Create the SSL_CONFIG/cards_certs directory and instruct the Docker build/startup process to load all certificates in this directory into the CARDS container's Java trusted CA certificate keystore
